bobtimus/src/loan.rs

```diff
@@ -110,24 +113,29 @@ pub fn loan_calculation_and_validation(
     price_fluctuation_interval: (Decimal, Decimal),
     current_price: LiquidUsdt,
 ) -> Result<ValidatedLoan> {
-    let interest_rate = calculate_interest_rate(
-        loan_request.term,
-        loan_request.collateralization,
-        &loan_offer.terms,
-        &loan_offer.collateralizations,
-        loan_offer.base_interest_rate,
-    )?;
+    let selected_term = match loan_offer
```

Comment:
This is validated double now - there is still a check in `validate_loan_is_acceptable` for the term. 
My idea was to keep **all** the validation steps in one place and accept that there are some calculations that we might do beforehand even though the loan is not acceptable. 

With introducing this early exit scenario we have to maintain this validation step twice in code. 
Either we remove it here again, or we remove it in the validation step. If we remove the validation step we split up the validation again and don't contain it in one function.